### PR TITLE
Add AssertOnlyDeps variant

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

File: depcheck/depcheck.go [hunk 1]
```diff
@@ -109,3 +109,32 @@ func AssertNoDependency(t *testing.T, banned map[string][]string) {
...
+			allowed[x] = struct{}{}
+		}
+		t.Run(ip, func(t *testing.T) {
+			g, err := buildGraph(ip)
+			if err != nil {
+				t.Fatal("buildGraph(queue) =", err)
+			}
+			for name := range g {
+				if _, ok := allowed[name]; !ok {
+					t.Errorf("dependency of %s not explicitly allowed %s\n%s", ip, name,
```
Comment (vagababov) [inline depcheck/depcheck.go:134]:
```suggestion
					t.Errorf("Dependency %s of %s is not explicitly allowed %s\n%s", name, ip,
```
I think is more intuitive.

**Review by vagababov — COMMENTED** [review]
In general, I think the core functionality should be extracted into a bool returning function so that the unit tests can test both positive and negative coverage. Right now we don't know if the failures work, besides manual checking.

**Comment by mattmoor** [comment]
We can add `Check` variants that return an `error`, but I think a `bool` is far too lossy.

Probably worth doing separately.

**Comment by vagababov** [comment]
OK I can do that :)

/lgtm

**Pull request merged**